tests/acceptance/run.sh

```diff
@@ -0,0 +1,672 @@
+	# the remaining number of suites that need to be distributed (could be zero)
+	REMAINING_SUITES=$((${COUNT_ALL_SUITES} - (${DIVIDE_INTO_NUM_PARTS} * ${MIN_SUITES_PER_RUN})))
+
+	if [[ ${RUN_PART} -le ${REMAINING_SUITES} ]]
+	then
+		SUITES_THIS_RUN=${MAX_SUITES_PER_RUN}
+		SUITES_IN_PREVIOUS_RUNS=$((${MAX_SUITES_PER_RUN} * (${RUN_PART} - 1)))
+	else
+		SUITES_THIS_RUN=${MIN_SUITES_PER_RUN}
+		SUITES_IN_PREVIOUS_RUNS=$((((${MAX_SUITES_PER_RUN} * ${REMAINING_SUITES}) + (${MIN_SUITES_PER_RUN} * (${RUN_PART} - ${REMAINING_SUITES} - 1)))))
+	fi
+
+	if [ ${SUITES_THIS_RUN} -eq 0 ]
+	then
+		echo "there are only ${COUNT_ALL_SUITES} suites, nothing to do in part ${RUN_PART}"
+		exit 0
+	fi
+
+	COUNT_FINISH_AND_TODO_SUITES=$((${SUITES_IN_PREVIOUS_RUNS} + ${SUITES_THIS_RUN}))
+	BEHAT_SUITES+=(`echo "${ALL_SUITES}" | head -n ${COUNT_FINISH_AND_TODO_SUITES} | tail -n ${SUITES_THIS_RUN}`)
+}
+
+if [[ -n "${BEHAT_SUITE}" ]]
+then
+	BEHAT_SUITES+=("${BEHAT_SUITE}")
+else
+	if [[ -n "${RUN_PART}" ]]; then
+		if [[ "${ACCEPTANCE_TEST_TYPE}" == "core-api" ]]; then
+			get_behat_suites "core"
+		else
+			get_behat_suites "${ACCEPTANCE_TEST_TYPE}"
+		fi
+	fi
+fi
+
+TEST_TYPE_TAG="@api"
+TEST_TYPE_TEXT="API"
+
+# Always have "@api"
+if [ -z "${BEHAT_FILTER_TAGS}" ]
+then
+	BEHAT_FILTER_TAGS="${TEST_TYPE_TAG}"
+else
+	# Be nice to the caller
+	# Remove any extra "&&" at the end of their tags list
+	BEHAT_FILTER_TAGS="${BEHAT_FILTER_TAGS%&&}"
+	# Remove any extra "&&" at the beginning of their tags list
+	BEHAT_FILTER_TAGS="${BEHAT_FILTER_TAGS#&&}"
+	BEHAT_FILTER_TAGS="${BEHAT_FILTER_TAGS}&&${TEST_TYPE_TAG}"
+fi
+
+# EMAIL_HOST defines where the system-under-test can find the email server (inbucket)
+# for sending email.
+if [ -z "${EMAIL_HOST}" ]
+then
+	EMAIL_HOST="127.0.0.1"
+fi
+
+# LOCAL_INBUCKET_HOST defines where this test script can find the Inbucket server
+# for sending email. When testing a remote system, the Inbucket server somewhere
+# "in the middle" might have a different host name from the point of view of
+# the test script.
+if [ -z "${LOCAL_EMAIL_HOST}" ]
+then
+	LOCAL_EMAIL_HOST="${EMAIL_HOST}"
+fi
+
+if [ -z "${EMAIL_SMTP_PORT}" ]
+then
+	EMAIL_SMTP_PORT="2500"
+fi
+
+# If the caller did not mention specific tags, skip the skipped tests by default
+if [ "${BEHAT_TAGS_OPTION_FOUND}" = false ]
+then
+	# If the caller has already specified specifically to run "@skip" scenarios
+	# then do not append "not @skip"
+	if [[ ! ${BEHAT_FILTER_TAGS} =~ "&&@skip&&" ]]
+	then
+		BEHAT_FILTER_TAGS="${BEHAT_FILTER_TAGS}&&~@skip"
+	fi
+fi
+
+export IPV4_URL
+export IPV6_URL
+export FILES_FOR_UPLOAD="${SCRIPT_PATH}/filesForUpload/"
+
+if [ "${TEST_OCIS}" != "true" ] && [ "${TEST_REVA}" != "true" ]
+then
+	# We are testing on an ownCloud core server.
+	# Tell the tests to wait 1 second between each upload/delete action
+	# to avoid problems with actions that depend on timestamps in seconds.
+	export UPLOAD_DELETE_WAIT_TIME=1
+fi
```

Comment:
This bit of code might now be useless here in ocis.
But maybe it will be needed (and the if condition modified a bit) if we enable the parallel deployment tests.
So we can leave it here for now.